src/screens/CoursesScreen.tsx

```diff
@@ -17,7 +17,10 @@ const CoursesScreen: React.FC<CoursesScreenProps> = ({ navigation }) => {
   const [error, setError] = useState<string | null>(null);
 
   useEffect(() => {
+    // Fetch courses everytime the screen is in focus
+    const unsubscribe = navigation.addListener('focus', fetchCourses);
     fetchCourses();
+    return unsubscribe;
   }, []);
```

Comment:
I think we should keep using useFocusEffect instead of useEffect with an event listener like this. It is usually recommended when working with React Navigation.

Reply:
Alright, gonna change it tomorrow.

Reply:
Also didn't realize there are merge conflicts. I will resolve this issue tomorrow too.

Reply:
I've already included it in the last commit. You should resolve the conflicts and it would be good.

Reply:
Conflict has been resolved.